### DISCUSSION TITLE
Rocket League

### DISCUSSION THREAD

**Discussion by 503geek** [nefarius/HidHide#32] [Q&A] [answered]
The docs mention Rocket League picking up controller 1 as an additional controller (thus creating a local party member) and blocking normal gamepad input.

I have this problem *only after Windows 10 64 bit (21H1) resumes from sleep state*. Everything works after a reboot, and I haven't found another game that has this issue (though I haven't been playing many local co-op games lately).

I'm using a genuine DS3 with DS4Windows. This issue does not occur if use SXS in DsHidMini, but that isn't a viable solution for me since I get weird input lag and I'd rather not use the Steam Configurator for the controller.

If this is a known issue, I am a developer and I may be able to help patch it.

I'm using a wired USB connection. I don't have BT. The controller LEDs are off after the wake from sleep and I have to unplug it and plug it back in to get HidHide and DS4 to see it. What is interesting is even with the controller powered off, Windows still thinks a device is being disconnected when I unplug it (it plays the USB device disconnection sound). I haven't been able to find the device it thinks is still connected. I have HidGuardHelper on my computer but it doesn't seem to do anything. HidGuardian is not installed (I verified that with the Installer from Google Drive).

I definitely only have one gaming device connected at all times. I have used XBOX 360 controllers in the past, and the drivers for those are still installed, but they are not connected at this time.

I have tried every possible combination of settings that I can think of:

*Whitelisting RL or not in HidHide
*Whitelisting DS4 (or not) in HidHide
*Adding a profile for RL in DS4 or not
*Hiding or not hiding the controller in HidHide and/or DS4
*Using shared or exclusive mode (or Dynamic or Permanent profiles) in DS4
*Various checkbox (red lock) states in HidHide for different devices

In total I've launched the game about 40 times with different configurations and the only thing that I've found that fixes it is a reboot.

I suppose it is possible that default settings would've worked and I just didn't know what I was doing when I first installed everything, and by trying to prevent the issue, I may have caused it. But I don't think that is the case. I didn't really change many settings until after I noticed the issue after the first resume from sleep.

Here's a screenshot of my most recent DS4 log

![image](https://user-images.githubusercontent.com/2246478/123336606-f4dd7a00-d4fa-11eb-9807-0bae0a082ff2.png)

The issue occurred after the second start up at 2:27pm. I didn't find anything that I thought was useful under Nefarius in the Windows event viewer.

**Comment by Kanuan** [comment, marked as answer]
This is probably not HidHide related, but anyway, when the problem happens again:
- Press `Win + R` -> Type `joy.cpl` then select OK -> Send a screenshot of the appearing window (Windows' `USB Game Controllers`)
- On DS4Windows `Settings` tab, select `Xinput Checker` and check which Slots are marked as connected

**Reply by 503geek** [reply]
![image](https://user-images.githubusercontent.com/2246478/123340018-71268c00-d500-11eb-956e-1e71fd25d8ef.png)

I joined the Discord and I see that this has come up on there already. Sorry for the bother. XInput Checker shows it as Slot 2 and see that this is a general Windows issue. Sounds like what you are doing is using BT and the controllers disconnect before the PC sleeps and then reconnect as you would expect them to. I believe I have a BT dongle somewhere and I will try testing that. Thank you for your time!

**Reply by Kanuan** [reply]
Yeah, Rocket League needs the controller to be in XInput Slot 1.
You can use [process explorer](https://docs.microsoft.com/en-us/sysinternals/downloads/process-explorer) to investigate what is keeping Slot 1 alive (I always bet on Chrome) 
![image](https://user-images.githubusercontent.com/24910442/123340694-05a1e600-d523-11eb-9e83-8d90ea1738f4.png)
